### PR TITLE
fix(number-input): numbers getting truncated

### LINKED ISSUE #180: Number-Input: Numbers getting truncated
Hi,

we observed a weird bug when removing the last digit from a number.

**Steps to reproduce:**

1. Make sure, that decimal places and max value are not set.
2. Type for example "5000000"
3. Tab to the next element and back to the Number Input
4. Remove the last digit

As you can see, the value printed down below shows a different number.  In this case, it is "5,000". So, two to three zeros getting lost.

bug

### PR DESCRIPTION
**Pull request by schoenj**
fix #180

### CODE REVIEW

File: projects/components/number-input/src/number-input.component.ts [hunk 1]
```diff
@@ -477,4 +478,16 @@ export class PsNumberInputComponent extends _PsNumberInputMixinBase
       this._onModelTouched();
     }
   }
+
+  private _replaceAll(val: string, searchValue: string, replaceValue: string): string {
```
Comment (saithis) [inline projects/components/number-input/src/number-input.component.ts:482]: I would place this in utils

**Review by saithis — APPROVED** [review]

File: projects/components/number-input/src/number-input.component.ts (hunk 1 above)
Comment (schoenj) [reply projects/components/number-input/src/number-input.component.ts]: done

**Comment by schoenj** [comment]
I changed the replaceAll implementation, so it uses split and join, because somehow the while-loop with the indexOf causes chrome to crash during tests.
When running the created test only or debugging it, i can go through step by step and it works.

I found other possible implementations as well, but none of these work properly or look quite hacky.
@saithis Is there a better solution?

File: projects/components/utils/src/string.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export function replaceAll(val: string, searchValue: string, replaceValue: string): string {
```
Comment (saithis) [inline projects/components/utils/src/string.ts:1]: tests would be nice :)

File: projects/components/utils/src/string.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  // .replace interprets searchValue as a string and will only replace the first occurnce.
+  // .replaceAll was added in ECMA-262 (2021) but is not currently available in nodeJS.
+
+  // https://stackoverflow.com/questions/1144783/how-to-replace-all-occurrences-of-a-string-in-javascript/54414146
+  // Solutions that do not work properly
+  // - while indexOf() > -1: Lets somehow crash chrome during tests
+  // - new Regex(searchValue, 'g'): Does not work because the dot and other characters would need to be escaped first
+  // - Escaping searchValue and then new Regex(...): Does not look like a proper solution either
+
+  val = val.split(searchValue).join(replaceValue);
```
Comment (saithis) [inline projects/components/utils/src/string.ts:11]: should we handle null/undefined? what happens when native replace receives null?

Comment (saithis) [reply projects/components/utils/src/string.ts:11]: oh never mind, nullvar.replace(...) obviously throws^^

**Review by saithis — APPROVED** [review]

**Pull request merged**